File: Arrays_n_Strings/longest_sliding_window.js
/* Finding the longest length of a sliding window with a targeted sum */

function findLength(arr, targetSum) {
    let left = 0;
    let currentSum = 0;
    let answer = 0;

    for ( let right = 0; right < arr.length; right++) {
        currentSum += arr[right];
        while (currentSum > targetSum) {
            currentSum -= arr[left];
            left++;
        }

        answer =  Math.max(answer, right - left + 1);
    }
    return answer;
}

let arr = [3, 1, 2, 7, 4, 2, 1, 1, 5];
let targetSum = 8;
 
console.log(findLength(arr, targetSum));

File: Arrays_n_Strings/longest_sliding_window.py
def find_length(arr, target_sum):
    left = 0
    current_sum = 0
    answer = 0

    for right in range(len(arr)):
        current_sum += arr[right]
        while current_sum > target_sum:
            current_sum -= arr[left]
            left += 1

        answer = max(answer, right - left + 1)
    
    return answer

arr = [3, 1, 2, 7, 4, 2, 1, 1, 5]
target_sum = 8

print(find_length(arr, target_sum))
